src/toil/test/provisioners/clusterScalerTest.py

```diff
@@ -153,47 +159,113 @@ def testClusterScaling(self):
         config.defaultDisk = 1
 
         # No preemptable nodes/jobs
-        config.maxPreemptableNodes = 0  # No preemptable nodes
+        config.maxPreemptableNodes = []  # No preemptable nodes
 
         # Non-preemptable parameters
-        config.nodeType = Shape(20, 10, 10, 10)
-        config.minNodes = 0
-        config.maxNodes = 10
+        config.nodeTypes = [Shape(20, 10, 10, 10, False)]
+        config.minNodes = [0]
+        config.maxNodes = [10]
 
         # Algorithm parameters
         config.alphaPacking = 0.8
         config.betaInertia = 1.2
         config.scaleInterval = 3
 
-        self._testClusterScaling(config, numJobs=100, numPreemptableJobs=0)
+        self._testClusterScaling(config, numJobs=100, numPreemptableJobs=0, jobShape=config.nodeTypes[0])
+        
+    def testClusterScalingMultipleNodeTypes(self):
+
+        smallNode = Shape(20, 5, 10, 10, False)
+        mediumNode = Shape(20, 10, 10, 10, False)
+        largeNode = Shape(20, 20, 10, 10, False)
+
+        numJobs = 100
+        
+        config = Config()
+
+        # Make defaults dummy values
+        config.defaultMemory = 1
+        config.defaultCores = 1
+        config.defaultDisk = 1
+
+        # No preemptable nodes/jobs
+        config.preemptableNodeTypes = []
+        config.minPreemptableNodes = []
+        config.maxPreemptableNodes = []  # No preemptable nodes
+
+        #Make sure the node types don't have to be ordered
+        config.nodeTypes = [largeNode, smallNode, mediumNode]
+        config.minNodes = [0, 0, 0]
+        config.maxNodes = [10, 10, 10]
+
+        # Algorithm parameters
+        config.alphaPacking = 0.8
+        config.betaInertia = 1.2
+        config.scaleInterval = 3
+        
+        mock = MockBatchSystemAndProvisioner(config, secondsPerJob=2.0)
+        clusterScaler = ClusterScaler(mock, mock, config)
+        clusterScaler.start()
+        mock.start()
+
+        try:
+            #Add small jobs
+            map(lambda x: mock.addJob(jobShape=smallNode), range(numJobs))
+            map(lambda x: mock.addJob(jobShape=mediumNode), range(numJobs))
```

Comment:
This is what broke the last build, should be list(map(

Reply:
OK. Thanks @adderan